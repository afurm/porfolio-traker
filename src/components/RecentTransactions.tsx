import React from 'react';
import { Box, Typography, List, ListItem, ListItemText, Divider, Paper } from '@mui/material';

interface Transaction {
  coinName: string;
  amount: number;
  type: 'buy' | 'sell';
  date: string;
}

export interface RecentTransactionsProps {
  transactions: Transaction[];
}

const RecentTransactions: React.FC<RecentTransactionsProps> = ({ transactions }) => {
  return (
    <Box sx={{ p: 4, maxWidth: 600, mx: 'auto' }}>
      <Paper elevation={3} sx={{ p: 3, borderRadius: 2 }}>
        <Typography variant="h5" fontWeight="bold" textAlign="center" mb={3}>
          Recent Transactions
        </Typography>
        {transactions.length === 0 ? (
          <Typography textAlign="center" variant="body1" color="textSecondary">
            No recent transactions found
          </Typography>
        ) : (
          <List>
            {transactions.map((transaction, index) => (
              <React.Fragment key={index}>
                <ListItem>
                  <ListItemText
                    primary={`${transaction.type === 'buy' ? 'Bought' : 'Sold'} ${transaction.amount} ${transaction.coinName}`}
                    secondary={transaction.date}
                  />
                </ListItem>
                {index < transactions.length - 1 && <Divider />}
              </React.Fragment>
            ))}
          </List>
        )}
      </Paper>
    </Box>
  );
};

export default RecentTransactions;
