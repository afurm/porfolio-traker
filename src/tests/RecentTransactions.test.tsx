import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import RecentTransactions, { RecentTransactionsProps } from '@/components/RecentTransactions';

describe('RecentTransactions Component Snapshot Test', () => {
  it('renders correctly with no transactions', () => {
    const mockProps: RecentTransactionsProps = {
      transactions: [],
    };

    const { asFragment } = render(<RecentTransactions {...mockProps} />);

    // Take a snapshot of the initial rendering with no transactions
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with some transactions', () => {
    const mockProps: RecentTransactionsProps = {
      transactions: [
        {
          coinName: 'Bitcoin',
          amount: 1.5,
          type: 'buy',
          date: '2024-10-05',
        },
        {
          coinName: 'Ethereum',
          amount: 2,
          type: 'sell',
          date: '2024-10-06',
        },
        {
          coinName: 'Dogecoin',
          amount: 1000,
          type: 'buy',
          date: '2024-10-07',
        },
      ],
    };

    const { asFragment } = render(<RecentTransactions {...mockProps} />);

    // Take a snapshot of the rendering with some transactions
    expect(asFragment()).toMatchSnapshot();
  });
});
