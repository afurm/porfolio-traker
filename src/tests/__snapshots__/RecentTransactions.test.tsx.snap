// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentTransactions Component Snapshot Test renders correctly with no transactions 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-15mclam"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-yt9ce0-MuiPaper-root"
      style="--Paper-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1ffs2s2-MuiTypography-root"
      >
        Recent Transactions
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1vfzsi2-MuiTypography-root"
      >
        No recent transactions found
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`RecentTransactions Component Snapshot Test renders correctly with some transactions 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-15mclam"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-yt9ce0-MuiPaper-root"
      style="--Paper-shadow: 0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12);"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1ffs2s2-MuiTypography-root"
      >
        Recent Transactions
      </h5>
      <ul
        class="MuiList-root MuiList-padding css-1u169s7-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-193q0wg-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-b8n15h-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
            >
              Bought 1.5 Bitcoin
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-ihhumc-MuiTypography-root"
            >
              2024-10-05
            </p>
          </div>
        </li>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-193q0wg-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-b8n15h-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
            >
              Sold 2 Ethereum
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-ihhumc-MuiTypography-root"
            >
              2024-10-06
            </p>
          </div>
        </li>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-193q0wg-MuiListItem-root"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-b8n15h-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
            >
              Bought 1000 Dogecoin
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-ihhumc-MuiTypography-root"
            >
              2024-10-07
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
